File: elixir/ping_pong.exs
defmodule PingPong do
  def ping(0) do
    send(:pong_process, :finished)
    IO.puts("Ping finished")
  end

  def ping(n) do
    send(:pong_process, :ping)

    receive do
      :pong ->
        IO.puts("Ping received pong")
    end

    ping(n - 1)
  end

  def pong do
    receive do
      :finished ->
        IO.puts("Pong finished")

      :ping ->
        IO.puts("Pong received")
        send(:ping_process, :pong)
        pong()
    end
  end

  def start do
    Process.register(spawn(fn -> pong() end), :pong_process)
    Process.register(spawn(fn -> ping(3) end), :ping_process)
  end
end

File: erlang/ping_pong.erl
-module(ping_pong).

-export([start/0, ping/1, pong/0]).

ping(0) ->
    pong_process ! finished,
    io:format("Ping finished~n", []);

ping(N) ->
    pong_process ! ping,
    receive
        pong ->
            io:format("Ping received pong~n", [])

    end,
    ping(N - 1).

pong() ->
    receive
        finised ->
            io:format("Pong finised~n", []);

        ping ->
            io:format("Pong received~n", []),
            ping_process ! pong,
            pong()
    end.

start() ->
    register(pong_process, spawn(ping_pong, pong, [])),
    register(ping_process, spawn(ping_pong, ping, [3])).